hub/api/dataset.py

```diff
@@ -151,17 +151,16 @@ def _get_dictionary(self, subpath, slice_=None):
                     if split_key[i] not in cur.keys():
                         cur[split_key[i]] = {}
                     cur = cur[split_key[i]]
-                slice_ = slice_ if slice_ else slice(0, self.shape[0])
+                slice_ = slice_ or slice(0, self.shape[0])
                 cur[split_key[-1]] = TensorView(
                     dataset=self, subpath=key, slice_=slice_
                 )
-        if len(tensor_dict) == 0:
+        if not tensor_dict:
             raise KeyError(f"Key {subpath} was not found in dataset")
         return tensor_dict
 
     def __iter__(self):
-        for i in range(len(self)):
-            yield self[i]
+        yield from self
```

Comment:
Because of this, one of the tests failed and we received 

```python
hub/compute/pipeline.py:24: in store
    results = [self._func(item) for item in self._ds]
hub/compute/pipeline.py:24: in <listcomp>
    results = [self._func(item) for item in self._ds]
hub/api/dataset.py:163: in __iter__
    yield from self
hub/api/dataset.py:163: in __iter__
    yield from self
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
```